src/capstone2llvmir/capstone2llvmir_impl.h

```diff
@@ -161,29 +161,34 @@ class Capstone2LlvmIrTranslator_impl : virtual public Capstone2LlvmIrTranslator
 			THROW,
 			/// Operand types does not have to be equal.
 			NOTHING,
-			/// Convert to destination type using ZEXT or TRUNC.
-			/// Types must be integer, or LLVM asserts.
-			ZEXT_TRUNC,
-			/// Convert to destination type using SEXT or TRUNC.
-			/// Types must be integer, or LLVM asserts.
-			SEXT_TRUNC,
-			/// Convert to destination type using FPCast (FPExt, BitCast,
+			/// Convert to destination integer type using ZEXT or TRUNC.
+			/// If source is FP type converts it using bitcast.
+			ZEXT_TRUNC_OR_BITCAST,
+			/// Convert to destination integer type using SEXT or TRUNC.
+			/// If source is FP type converts it using bitcast.
+			SEXT_TRUNC_OR_BITCAST,
+			/// Convert to destination FP type using FPCast (FPExt, BitCast,
 			/// or FPTrunc).
-			/// Types must be floating point, or LLVM asserts.
-			FP_CAST,
-			/// Convert to destination type using SIToFP.
+			/// If source is integer type converts it using bitcast.
+			FPCAST_OR_BITCAST,
+			/// Convert to destination FP type using SIToFP.
 			/// Source must be integer, destination fp, or LLVM asserts.
-			SITOFP,
-			/// Convert to destination type using UIToFP.
+			SITOFP_OR_FPCAST,
+			/// Convert to destination FP type using UIToFP.
 			/// Source must be integer, destination fp, or LLVM asserts.
-			UITOFP
+			UITOFP_OR_FPCAST
 		};
 
 		llvm::Value* generateTypeConversion(
 				llvm::IRBuilder<>& irb,
 				llvm::Value* from,
 				llvm::Type* to,
 				eOpConv ct);
+
+		llvm::Type* checkTypeConversion(
```

Comment:
What does this do? I see that it is used only in `_loadOps()`. There is no comment for the method. I don't like it much, because judging only by the name, it looks like something that should be called before/after(?) every type conversion (`generateTypeConversion()`). This would confuse me.

Reply:
This is meant to be only an internal method for the `_loadOps()` method. 

A request for this method originated in how `_loadOps()` is used. As the size of the type of loaded operands of translated instructions may vary from usage to usage, this method cannot be provided with specific `llvm::Type*` for loaded operands. Translation methods, however, know whether the full type will be integer type or floating-point type, and so the only information that they'll provide to the method `_loadOps()` is the conversion type. Thus it has to be able to deduce the type from the conversion method and the size of the type from the first loaded operand.

This deduction is provided by the `checkTypeConvertion` method that will be supplied with type and will return it as a prototype with a known size that can be used to convert other operands. If the type does not correspond to the used conversion method,  however, it tries to create type with equivalent size (either FP or integer - based on conversion type) and returns this type as a prototype for nest conversions.

Consider this example from issue 633:

The translation method of the instruction `ORI` receives two operands `A = Float128` and  `B = Float128`.
The translation method knows that its operands can only have integer type as it operates only on bits. This method does not care about the sign of integer operands. Conversion method that will ensure that every operand has equal size would be `ZEXT_OR_TRUNC`, and if we consider that by some mistake, a floating-point register may get into the input of this method, `ZERO_TRUNC_OR_BITCAST` will do the job. So this method uses `_loadOps` to load two integer operands of the `ORI` instruction. The method `_loadOps` now must deduce the size of operands, for example, by loading the first one.

Before this PR, the method has taken the first operand and declared its type as the result type. Other operands would get converted to this result type. As the type of first operands is Float128, this would result in incorrect type guessed as result type.

Now, the `_loadOps` method uses `checkTypeCovnertion` that will check whether it is possible to use the type of the first operand as example type (result type) or wheater it should be other types with the same size. So a combination of ZEXT_TRUNC_OR_BITCAST + Flaot128 operand will result in result type to be Int128, and both operands (A, B) will be converted to this type.

Reply:
I will provide documentation for this method and mark it as internal with an underscore (`_checkTypeConvertion`).